### PR TITLE
[TimeSeriesInsights] `az tsi access-policy/environment/reference-data-set`: Migrate to Codegen V2

### PR DESCRIPTION
**Pull request by calvinhzy**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Timeseriesinsights

**Review by necusjz — APPROVED** [review]

File: src/timeseriesinsights/azext_timeseriesinsights/custom.py [hunk 1]
```diff
@@ -327,121 +290,31 @@ def timeseriesinsights_event_source_iot_hub_update(client,
...
-                                               environment_name,
-                                               reference_data_set_name):
-    return client.get(resource_group_name=resource_group_name,
-                      environment_name=environment_name,
-                      reference_data_set_name=reference_data_set_name)
+    List = _ReferenceDataSetList(cmd.loader)
+    parameters = {}
+    parameters['resource_group'] = resource_group_name
+    parameters['environment_name'] = environment_name
+    return List(parameters)["value"]
```
Comment (kairu-ms) [inline src/timeseriesinsights/azext_timeseriesinsights/custom.py:300]: Why not use list directly?

File: src/timeseriesinsights/azext_timeseriesinsights/custom.py [hunk 2]
```diff
@@ -42,61 +22,58 @@ def timeseriesinsights_environment_gen1_create(client,
...
 
 
-def timeseriesinsights_environment_gen1_update(client,
+def timeseriesinsights_environment_list(cmd,
+                                        resource_group_name=None):
+    List = _EnvironmentList(cmd.loader)
+    parameters = {}
+    parameters['resource_group'] = resource_group_name
+
+    return List(parameters)["value"]
```
Comment (kairu-ms) [inline src/timeseriesinsights/azext_timeseriesinsights/custom.py:47]: Same concern here

File: src/timeseriesinsights/azext_timeseriesinsights/custom.py (hunk 1 above)
Comment (calvinhzy) [reply src/timeseriesinsights/azext_timeseriesinsights/custom.py]: Need to get .value as the List command does not support pagination.

**Pull request merged**